mysite/blog/views.py

```diff
@@ -23,8 +37,27 @@ def post_detail(request, year, month, day, post):
         publish__month=month,
         publish__day=day,
     )
-    return render(request, "blog/post/detail.html", {"post": post})
+    comments = getCommentsoFPost(post)
+    new_comment = None
+
+
+    if request.method == "POST":
+        comment_form = CommentForm(data=request.POST)
+        new_comment=check_comment_form_validity_and_save_comment(comment_form,post)
+      
+    else:
+        comment_form = CommentForm()
```

Comment:
```
form = CommentForm(data=request.POST or None)
if form.is_valid():
    comment = form.save(commit=False)
    comment.post = post
    comment.save()
    messages.success("Comment added successfully")
    return HttpRedirect("post-detail")
    
return render(request, "blog/post/detail.html", {"post": post, "form": form})
```

Also read https://stackoverflow.com/questions/38240103/why-do-we-need-redirect-after-sending-post-request